featurebyte/service/feature_store_warehouse.py

```diff
@@ -54,78 +55,48 @@
     os.environ.get("MAX_TABLE_CELLS", 10000000 * 300)
 )  # 10 million rows, 300 columns
 
-DATABASE_CACHE_MAX_SIZE = 100
-DATABASE_CACHE_TTL_SECONDS = 3600
-
-
-def cache_factory() -> TTLCache[Any, Any]:
-    """
-    Factory function to create a TTLCache instance with specified max size and TTL.
-
-    Returns
-    -------
-    TTLCache[Any, Any]
-    """
-    return TTLCache(maxsize=DATABASE_CACHE_MAX_SIZE, ttl=DATABASE_CACHE_TTL_SECONDS)
-
-
-list_database_cache: Dict[ObjectId, TTLCache[Any, list[str]]] = defaultdict(cache_factory)
-list_schema_cache: Dict[ObjectId, TTLCache[Any, list[str]]] = defaultdict(cache_factory)
-list_table_cache: Dict[ObjectId, TTLCache[Any, list[TableSpec]]] = defaultdict(cache_factory)
-list_column_cache: Dict[ObjectId, TTLCache[Any, list[ColumnSpecWithDescription]]] = defaultdict(
-    cache_factory
-)
-table_details_cache: Dict[ObjectId, TTLCache[Any, TableDetails]] = defaultdict(cache_factory)
-
 logger = get_logger(__name__)
 
 
 T = TypeVar("T")
 
 
-def async_cache(
-    cache_dict: Dict[ObjectId, TTLCache[Any, Any]],
-) -> Callable[[Callable[..., Awaitable[T]]], Callable[..., Awaitable[T]]]:
-    """
-    Decorator for async cache lookup and storage that preserves return types.
-
-    Parameters
-    ----------
-    cache_dict: Dict[ObjectId, TTLCache[Any, Any]]
-        Dictionary to store the cache for different feature stores.
-
-    Returns
-    -------
-    Callable[[Callable[..., Awaitable[T]]], Callable[..., Awaitable[T]]]
-        Decorator function that wraps the original function while preserving return type.
-    """
+def async_cache(func: Callable[..., Awaitable[T]]) -> Callable[..., Awaitable[T]]:
+    @functools.wraps(func)
+    async def wrapper(
+        self: FeatureStoreWarehouseService,
+        feature_store: FeatureStoreModel,
+        *args: Any,
+        refresh: bool = True,
+        **kwargs: Any,
+    ) -> T:
+        credentials = await self.session_manager_service.credential_service.get_credentials(
+            user_id=self.session_manager_service.user.id, feature_store=feature_store
+        )
+        keys = hashkey(func.__name__, credentials.id if credentials else None, *args, **kwargs)
+        str_keys = [str(key) for key in keys]
```

Comment:
Is there a reason to keep a list of keys instead of hashing everything into a single string?